src/shared/components/data-table/components/table-controls/components/table-actions/components/download-data/DownloadData.tsx

```diff
@@ -0,0 +1,134 @@
+/**
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import React, { ReactNode, useEffect, useRef } from 'react';
+import ReactDOM from 'react-dom/client';
+import memoize from 'lodash/memoize';
+
+import useDataTable from 'src/shared/components/data-table/hooks/useDataTable';
+import { downloadTextAsFile } from 'src/shared/helpers/downloadAsFile';
+
+import LoadingButton from 'src/shared/components/loading-button';
+
+import { TableAction } from 'src/shared/components/data-table/dataTableTypes';
+
+import styles from './DownloadData.scss';
+
+const getReactRenderer = memoize(() => {
+  const element = document.createElement('div');
+  const root = ReactDOM.createRoot(element);
+
+  return {
+    element,
+    renderer: root
+  };
+});
+
+const getReactNodeText = (node: ReactNode): string => {
+  const { element, renderer } = getReactRenderer();
+  renderer.render(node);
+  return element.innerText;
+};
+
+const DownloadData = () => {
+  const {
+    dispatch,
+    rows,
+    downloadFileName,
+    columns,
+    selectedAction,
+    hiddenRowIds,
+    downloadHandler
+  } = useDataTable();
+
+  const allowComponentResetRef = useRef(true);
+
+  useEffect(() => {
+    return () => {
+      allowComponentResetRef.current = false;
+    };
+  }, []);
+
+  const restoreDefaults = () => {
+    setTimeout(() => {
+      if (allowComponentResetRef.current) {
+        dispatch({
+          type: 'set_selected_action',
+          payload: TableAction.DEFAULT
+        });
+      }
+    }, 1000);
+  };
```

Comment:
Sorry, we messed up. There is now a second's delay when you click on "Cancel", which shouldn't be there.

https://user-images.githubusercontent.com/6834224/195531427-f7f9d940-a525-4903-86d3-7ff7e3e0d025.mov

Reply:
This is now fixed.